### PR TITLE
CU-yrnj0u Fix binary comp of 0.13.2

### PR DESCRIPTION
**Pull request by i-walker**
Minor binary incompatibilities:
- `raceN` `f*` functions have the CoroutineScope as a receiver
- Iterable functions are now in `Iterable.kt` and are moved out of the data type files, like `Option.kt`, `ParTraverse`,

### CODE REVIEW

File: arrow-libs/fx/arrow-fx-coroutines/api/arrow-fx-coroutines.api [hunk 1]
```diff
@@ -402,22 +443,22 @@ public final class arrow/fx/coroutines/Schedule$Companion {
 	public final fun duration ()Larrow/fx/coroutines/Schedule;
 	public final fun exponential (DD)Larrow/fx/coroutines/Schedule;
 	public static synthetic fun exponential$default (Larrow/fx/coroutines/Schedule$Companion;DDILjava/lang/Object;)Larrow/fx/coroutines/Schedule;
-	public final fun exponential-VtjQ1oo (DD)Larrow/fx/coroutines/Schedule;
```
[the remark this answers is not in the document]
Comment (nomisRev) [reply arrow-libs/fx/arrow-fx-coroutines/api/arrow-fx-coroutines.api:405]: I'm not sure. Seems like random identifiers. Can't find anything in their docs about this.

**Review by nomisRev — COMMENTED** [review]
Besides that, I think it looks good! No other breakages I could spot.

File: arrow-libs/core/arrow-core/api/arrow-core.api [hunk 2]
```diff
@@ -819,9 +854,8 @@ public final class arrow/core/OptionKt {
 	public static final fun sequence (Larrow/core/Option;)Ljava/util/List;
 	public static final fun sequenceEither (Larrow/core/Option;)Larrow/core/Either;
 	public static final fun sequenceValidated (Larrow/core/Option;)Larrow/core/Validated;
-	public static final fun singleOrNone (Ljava/lang/Iterable;)Larrow/core/Option;
-	public static final fun singleOrNone (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Larrow/core/Option;
```
[the remark this answers is not in the document]
Comment (i-walker) [reply arrow-libs/core/arrow-core/api/arrow-core.api:823]: I see I'll do that

**Review by nomisRev — APPROVED** [review]
- `arrow-xxx-test` we're ignorning for now. Api baseline will be tracked from 1.0.0

On the `0.13.3` release branch we can move the `Option/Iterable` methods back, and then we don't have a binary compat break between `0.13.2` and `0.13.3`.
This way the binary breakage is only between `0.13.3` and `1.0.0` for those methods, whilst the Kotlin API is never broken.

Looks great to me @i-walker!! Thank you so much for taking care of this!!

**Review by raulraja — APPROVED** [review]
Thanks @i-walker !

**Comment by i-walker** [comment]
Ok I move those files back and then we can merge this :)

**Pull request merged**